google/cloud/bigquery/v2/minimal/internal/table_partition.h

```diff
@@ -45,9 +45,9 @@ void to_json(nlohmann::json& j, TimePartitioning const& t);
 void from_json(nlohmann::json const& j, TimePartitioning& t);
 
 struct Range {
-  std::string start;
-  std::string end;
-  std::string interval;
+  int start;
```

Comment:
Do you know if these are 32-bit integers or 64-bit integers?  I suspect 64-bits because they represent row numbers and you can have more than $2^{32}$ rows.  So the correct type is `std::int64_t` or `std::uint64_t`.

Please remember that JSON represents integers larger than (or that can be larger than) $2^{52}$ as strings.

Reply:
I see you point.
I made some more tests and seems like `"0"` is also a valid parameter.
Unfortunatelly, it's valid only for `start` field, for `end` and `interval` in should be more than 0, at least `"1"`.
Let me close this PR now. I'll think how to eliminate this entire object if none of values were set by user.